crates/hdk/src/entry/delete_entry.rs

```diff
@@ -1,13 +1,12 @@
-/// Alias to delete!
+use crate::prelude::*;
+
+/// Alias to delete
 ///
 /// Takes any expression that evaluates to the HeaderHash of the deleted element.
 ///
 /// ```ignore
-/// delete_entry!(entry_hash!(foo_entry)?)?;
+/// delete_entry(entry_hash(foo_entry)?)?;
 /// ```
-#[macro_export]
-macro_rules! delete_entry {
-    ( $hash:expr ) => {{
-        delete!($hash)
-    }};
+pub fn delete_entry(hash: HeaderHash) -> HdkResult<HeaderHash> {
```

Comment:
I wonder if we actually want this to take an `EntryHash` and choose any header for that entry. Or we could make another function call `delete_any_entry(hash: EntryHash)`. It's just that deleting any entry will make `get` not return any entry regardless of `EntryHash` or `HeaderHash` as input. So this could match those semantics a bit better.
All good if this PR isn't the place for this.